### PR TITLE
CONTRIBUTING: add information about git and code structure

### LINKED ISSUE #1296: Improve CONTRIBUTING.md
I've been reading https://github.com/linux-system-roles/network/blob/main/contributing.md and I liked it a lot. The idea is to improve our current CONTRIBUTING.md file, including:

* [x] Include the code structure.
* [x] How to configure git, including a reference to Pro Git would be a great idea.
* [x] How to write a good commit message.
* [x] Include reference to our developer guide https://www.nmstate.io/devel/dev_guide.html
* [x] commit-template.txt is mentioned in CONTRIBUTING.md but does not seem to exist. Remove it please.

### PR DESCRIPTION
**Pull request by Razaloc**
- Add Code Structure.

 - Add Configure Git.

 - Add How to write a good commit.

 - Add references to Pro Git and the developer guide.

 - Erase reference to commit-template.txt

 - Erase empty file ./libnmstate/README.md

Signed-off-by: Rafael G. Ruiz <llerrak@hotmail.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -13,6 +13,54 @@ This project and everyone participating in it is governed by the
...
+
+- `./doc/` - Contains the documentation. 
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+interfaces. `./libnmstate/nispor/` Contains tools to comunicate with nispor. 
+`./libnmstate/nm/` Contains the support of NetworkManager. 
+`./libnmstate/plugins/` Contains the plugins supported by default (e.g. ovsdb). 
+`./libnmstate/schemas/` Contains the API.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: Can you break this line into smaller parts?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -13,6 +13,54 @@ This project and everyone participating in it is governed by the
...
+
+- `./nmstatectl/` - Contains command lines tools.
+
+- `./packaging/` - Contains packaging utilities.
+
+- `./tests/` - Contains tests for unitary integration, nmstatectl and the API.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: - `./tests/` - Contains tests for unit and integration tests.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -21,6 +74,8 @@ Please report unacceptable behavior to the nmstate team.
 - Code: Managed on [GitHub](https://github.com/nmstate/nmstate) through
   Pull Requests ([PR](https://github.com/nmstate/nmstate/pulls)).
 
+You can check the [developer guide](https://www.nmstate.io/devel/dev_guide.html).
```
Comment (ffmancera) [inline CONTRIBUTING.md]: I would prefer "Please check" instead of "You can check".

**Review by ffmancera — CHANGES_REQUESTED** [review]
I made a comment below. In addition, could you change the commit message to `CONTRIBUTING: add information about git and code structure`.

Other than that, it looks good. Thanks!

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -13,6 +13,59 @@ This project and everyone participating in it is governed by the
...
+- `./doc/` - Contains the documentation. 
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+interfaces.
+ 
+- `./libnmstate/nispor/` Contains tools to comunicate with nispor. 
+
+- `./libnmstate/nm/` Contains the support of NetworkManager.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: Contains the nmstate NetworkManager plugin, currently for querying network status and applying network states.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -13,6 +13,59 @@ This project and everyone participating in it is governed by the
...
+serving the tests of Nmstate.
+
+- `./doc/` - Contains the documentation. 
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+interfaces.
+ 
+- `./libnmstate/nispor/` Contains tools to comunicate with nispor.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: Contains the nmstate nispor plugin, currently for querying kernel network status.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -13,6 +13,59 @@ This project and everyone participating in it is governed by the
...
+- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+interfaces.
+ 
+- `./libnmstate/nispor/` Contains tools to comunicate with nispor. 
+
+- `./libnmstate/nm/` Contains the support of NetworkManager. 
+
+- `./libnmstate/plugins/` Contains the plugins supported by default (e.g. ovsdb). 
+
+- `./libnmstate/schemas/` Contains the API.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: Contains the API schema file.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -13,6 +13,59 @@ This project and everyone participating in it is governed by the
...
+
+- `./nmstatectl/` - Contains command lines tools.
```
Comment (cathay4t) [inline CONTRIBUTING.md]: Contains command lines tools and varlink service.

**Review by cathay4t — CHANGES_REQUESTED** [review]

**Review by cathay4t — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**